src/Cynthia.Card/src/Cynthia.Card.Server/Services/GwentGameService/GwentServerService.cs

```diff
@@ -368,40 +368,23 @@ public void InvokeGameOver(GameResult result, bool isOnlyShow, bool isCountMMR)
 
         public int CalculateMMR(int myMMR, int enemyMMR, bool isWin, bool isDraw)
         {
-            int k = CalculateK(myMMR);
-            double s = isDraw ? 0.5 : (isWin ? 1 : 0);
-            int diff = Math.Max(enemyMMR - myMMR, -50);
-            double e = 1 / (1.0 + Math.Pow(10, diff / 400.0));
-            double eta = CalculateEta(s, e, myMMR);
-            int addon = myMMR < 2000 ? (isWin ? 70 : 50) : 0;
-            return (int)Math.Round(myMMR + eta * k * (s - e) + addon);
+             int scale=2000;
+            int diff = enemyMMR - myMMR;
+            double result = isDraw ? 0.5 : (isWin ? 1 : 0);
+            double k = CalculateK(diff);
+            double expected = 1 / (1.0 + Math.Pow(10, diff / scale));
+            return (int)Math.Round(myMMR + k * (result - expected));
         }
 
-        public int CalculateK(int MMR)
+        public int CalculateK(int diff)
```

Comment:
Is there any reason you change the formula of k so heavily?

Reply:
i believe there is. If k is dependend on mmr value, then it can be different for both players - this can leed to infacion or even deflation of mmr (hard to predict). If u want to keep mmr changing slower at higher mmr (with i dont think is necessary) we can base is on average mmr of both players. my formula aims to lower the impact of games with high mmr diference - but to be honest its not mandatory. If you think its confusing k can be constant =40

Reply:
also - it would be fantastic if k could be higher for lets say first 50 games played after reset or smth for mmr to settle quicker, but I personally have no idea how to code that

Reply:
higher k doesnt mean you gain rating tho. It just means your rating is more prone to change - more volatile. That is why i think it would be perfect for k to be higher for new accounts.